<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Page not found | Malaria vaccine antigen identification for Plasmodium vivax</title>
  <meta name="description" content="Page not found | Malaria vaccine antigen identification for Plasmodium vivax" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Page not found | Malaria vaccine antigen identification for Plasmodium vivax" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Page not found | Malaria vaccine antigen identification for Plasmodium vivax" />
  
  
  

<meta name="author" content="Renee Ti Chou" />


<meta name="date" content="2023-06-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  


<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.27/datatables.js"></script>
<link href="libs/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.12.1/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="libs/dt-ext-fixedcolumns-1.12.1/css/fixedColumns.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-fixedcolumns-1.12.1/js/dataTables.fixedColumns.min.js"></script>
<link href="libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="libs/selectize-0.12.0/selectize.min.js"></script>
<!-- https://stackoverflow.com/questions/45360998/code-folding-in-bookdown -->
<script type="text/javascript" src="js/transition.js"></script>
<script type="text/javascript" src="js/dropdown.js"></script>
<script>
var str = '<div class="btn-group pull-right" style="position: fixed; right: 50px; top: 10px; z-index: 200"><button type="button" class="btn btn-default btn-xs dropdown-toggle" id="allCodeButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" data-_extension-text-contrast=""><span>Code</span> <span class="caret"></span></button><ul class="dropdown-menu" style="min-width: 50px;"><li><a id="rmd-show-all-code" href="#">Show All Code</a></li><li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li></ul></div>';
document.write(str);
</script>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "hide");
});
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="css/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./"><i>Positive-unlabeled learning for Pv RV</i></a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="id_01_data_engineering.html"><a href="id_01_data_engineering.html"><i class="fa fa-check"></i><b>1</b> Data engineering</a>
<ul>
<li class="chapter" data-level="1.1" data-path="id_01_data_engineering.html"><a href="id_01_data_engineering.html#purf-package-installization"><i class="fa fa-check"></i><b>1.1</b> <code>purf</code> package installization</a></li>
<li class="chapter" data-level="1.2" data-path="id_01_data_engineering.html"><a href="id_01_data_engineering.html#retrieving-p.-vivax-protein-variables"><i class="fa fa-check"></i><b>1.2</b> Retrieving <em>P. vivax</em> protein variables</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="id_01_data_engineering.html"><a href="id_01_data_engineering.html#immunological-data-sets"><i class="fa fa-check"></i><b>1.2.1</b> Immunological data sets</a></li>
<li class="chapter" data-level="1.2.2" data-path="id_01_data_engineering.html"><a href="id_01_data_engineering.html#proteomic-data-sets"><i class="fa fa-check"></i><b>1.2.2</b> Proteomic data sets</a></li>
<li class="chapter" data-level="1.2.3" data-path="id_01_data_engineering.html"><a href="id_01_data_engineering.html#structural-data-sets"><i class="fa fa-check"></i><b>1.2.3</b> Structural data sets</a></li>
<li class="chapter" data-level="1.2.4" data-path="id_01_data_engineering.html"><a href="id_01_data_engineering.html#genomic-data-sets"><i class="fa fa-check"></i><b>1.2.4</b> Genomic data sets</a></li>
<li class="chapter" data-level="1.2.5" data-path="id_01_data_engineering.html"><a href="id_01_data_engineering.html#final-assembly"><i class="fa fa-check"></i><b>1.2.5</b> Final assembly</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="id_01_data_engineering.html"><a href="id_01_data_engineering.html#generating-ml-input"><i class="fa fa-check"></i><b>1.3</b> Generating ML input</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="id_02_model_training.html"><a href="id_02_model_training.html"><i class="fa fa-check"></i><b>2</b> Model training</a>
<ul>
<li class="chapter" data-level="2.1" data-path="id_02_model_training.html"><a href="id_02_model_training.html#hyper-parameter-tuning-for-purf"><i class="fa fa-check"></i><b>2.1</b> Hyper-parameter tuning for PURF</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="id_02_model_training.html"><a href="id_02_model_training.html#analysis"><i class="fa fa-check"></i><b>2.1.1</b> Analysis</a>
<ul>
<li class="chapter" data-level="2.1.1.1" data-path="id_02_model_training.html"><a href="id_02_model_training.html#pv-data-set"><i class="fa fa-check"></i><b>2.1.1.1</b> Pv data set</a></li>
<li class="chapter" data-level="2.1.1.2" data-path="id_02_model_training.html"><a href="id_02_model_training.html#pv-pf-combined-data-set"><i class="fa fa-check"></i><b>2.1.1.2</b> Pv + Pf combined data set</a></li>
</ul></li>
<li class="chapter" data-level="2.1.2" data-path="id_02_model_training.html"><a href="id_02_model_training.html#plotting"><i class="fa fa-check"></i><b>2.1.2</b> Plotting</a>
<ul>
<li class="chapter" data-level="2.1.2.1" data-path="id_02_model_training.html"><a href="id_02_model_training.html#pv-data-set-1"><i class="fa fa-check"></i><b>2.1.2.1</b> Pv data set</a></li>
<li class="chapter" data-level="2.1.2.2" data-path="id_02_model_training.html"><a href="id_02_model_training.html#bimodal-distribution-plot"><i class="fa fa-check"></i><b>2.1.2.2</b> Bimodal distribution plot</a></li>
<li class="chapter" data-level="2.1.2.3" data-path="id_02_model_training.html"><a href="id_02_model_training.html#known-antigen-ranking"><i class="fa fa-check"></i><b>2.1.2.3</b> Known antigen ranking</a></li>
<li class="chapter" data-level="2.1.2.4" data-path="id_02_model_training.html"><a href="id_02_model_training.html#pv-pf-combined-data-set-1"><i class="fa fa-check"></i><b>2.1.2.4</b> Pv + Pf combined data set</a></li>
<li class="chapter" data-level="2.1.2.5" data-path="id_02_model_training.html"><a href="id_02_model_training.html#bimodal-distribution-plot-1"><i class="fa fa-check"></i><b>2.1.2.5</b> Bimodal distribution plot</a></li>
<li class="chapter" data-level="2.1.2.6" data-path="id_02_model_training.html"><a href="id_02_model_training.html#known-antigen-ranking-1"><i class="fa fa-check"></i><b>2.1.2.6</b> Known antigen ranking</a></li>
</ul></li>
<li class="chapter" data-level="2.1.3" data-path="id_02_model_training.html"><a href="id_02_model_training.html#results-for-positive-level-0.5"><i class="fa fa-check"></i><b>2.1.3</b> Results for positive level = 0.5</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="id_03_model_evaluation.html"><a href="id_03_model_evaluation.html"><i class="fa fa-check"></i><b>3</b> Model evaluation</a>
<ul>
<li class="chapter" data-level="3.1" data-path="id_03_model_evaluation.html"><a href="id_03_model_evaluation.html#known-antigen-prediction-accuracy"><i class="fa fa-check"></i><b>3.1</b> Known antigen prediction accuracy</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="id_03_model_evaluation.html"><a href="id_03_model_evaluation.html#analysis-1"><i class="fa fa-check"></i><b>3.1.1</b> Analysis</a>
<ul>
<li class="chapter" data-level="3.1.1.1" data-path="id_03_model_evaluation.html"><a href="id_03_model_evaluation.html#pv-model-cross-species-predictions"><i class="fa fa-check"></i><b>3.1.1.1</b> Pv model cross-species predictions</a></li>
<li class="chapter" data-level="3.1.1.2" data-path="id_03_model_evaluation.html"><a href="id_03_model_evaluation.html#pf-model-cross-species-predictions"><i class="fa fa-check"></i><b>3.1.1.2</b> Pf model cross-species predictions</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="id_03_model_evaluation.html"><a href="id_03_model_evaluation.html#adversarial-controls"><i class="fa fa-check"></i><b>3.2</b> Adversarial controls</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="id_03_model_evaluation.html"><a href="id_03_model_evaluation.html#analysis-2"><i class="fa fa-check"></i><b>3.2.1</b> Analysis</a>
<ul>
<li class="chapter" data-level="3.2.1.1" data-path="id_03_model_evaluation.html"><a href="id_03_model_evaluation.html#pv-data-set-2"><i class="fa fa-check"></i><b>3.2.1.1</b> Pv data set</a></li>
<li class="chapter" data-level="3.2.1.2" data-path="id_03_model_evaluation.html"><a href="id_03_model_evaluation.html#pv-pf-combined-data-set-2"><i class="fa fa-check"></i><b>3.2.1.2</b> Pv + Pf combined data set</a></li>
</ul></li>
<li class="chapter" data-level="3.2.2" data-path="id_03_model_evaluation.html"><a href="id_03_model_evaluation.html#plotting-1"><i class="fa fa-check"></i><b>3.2.2</b> Plotting</a>
<ul>
<li class="chapter" data-level="3.2.2.1" data-path="id_03_model_evaluation.html"><a href="id_03_model_evaluation.html#mean-differences-in-scores"><i class="fa fa-check"></i><b>3.2.2.1</b> Mean differences in scores</a></li>
<li class="chapter" data-level="3.2.2.2" data-path="id_03_model_evaluation.html"><a href="id_03_model_evaluation.html#investigation-of-known-antigen-groups"><i class="fa fa-check"></i><b>3.2.2.2</b> Investigation of known antigen groups</a></li>
<li class="chapter" data-level="3.2.2.3" data-path="id_03_model_evaluation.html"><a href="id_03_model_evaluation.html#labeled-known-antigen-accuracies"><i class="fa fa-check"></i><b>3.2.2.3</b> Labeled known antigen accuracies</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="id_04_model_comparisons.html"><a href="id_04_model_comparisons.html"><i class="fa fa-check"></i><b>4</b> Model comparisons</a>
<ul>
<li class="chapter" data-level="4.1" data-path="id_04_model_comparisons.html"><a href="id_04_model_comparisons.html#known-antigen-score-comparisons"><i class="fa fa-check"></i><b>4.1</b> Known antigen score comparisons</a></li>
<li class="chapter" data-level="4.2" data-path="id_04_model_comparisons.html"><a href="id_04_model_comparisons.html#known-antigen-prediction-summary"><i class="fa fa-check"></i><b>4.2</b> Known antigen prediction summary</a></li>
<li class="chapter" data-level="4.3" data-path="id_04_model_comparisons.html"><a href="id_04_model_comparisons.html#score-and-species-association"><i class="fa fa-check"></i><b>4.3</b> Score and species association</a></li>
<li class="chapter" data-level="4.4" data-path="id_04_model_comparisons.html"><a href="id_04_model_comparisons.html#tree-depth-analysis"><i class="fa fa-check"></i><b>4.4</b> Tree depth analysis</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="id_04_model_comparisons.html"><a href="id_04_model_comparisons.html#analysis-3"><i class="fa fa-check"></i><b>4.4.1</b> Analysis</a></li>
<li class="chapter" data-level="4.4.2" data-path="id_04_model_comparisons.html"><a href="id_04_model_comparisons.html#plotting-2"><i class="fa fa-check"></i><b>4.4.2</b> Plotting</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="id_05_model_interpretation.html"><a href="id_05_model_interpretation.html"><i class="fa fa-check"></i><b>5</b> Model interpretation</a>
<ul>
<li class="chapter" data-level="5.1" data-path="id_05_model_interpretation.html"><a href="id_05_model_interpretation.html#proximity-matrix-calculation"><i class="fa fa-check"></i><b>5.1</b> Proximity matrix calculation</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="id_05_model_interpretation.html"><a href="id_05_model_interpretation.html#analysis-4"><i class="fa fa-check"></i><b>5.1.1</b> Analysis</a>
<ul>
<li class="chapter" data-level="5.1.1.1" data-path="id_05_model_interpretation.html"><a href="id_05_model_interpretation.html#pv-data-set-3"><i class="fa fa-check"></i><b>5.1.1.1</b> Pv data set</a></li>
<li class="chapter" data-level="5.1.1.2" data-path="id_05_model_interpretation.html"><a href="id_05_model_interpretation.html#pv-pf-combined-data-set-3"><i class="fa fa-check"></i><b>5.1.1.2</b> Pv + Pf combined data set</a></li>
</ul></li>
<li class="chapter" data-level="5.1.2" data-path="id_05_model_interpretation.html"><a href="id_05_model_interpretation.html#plotting-3"><i class="fa fa-check"></i><b>5.1.2</b> Plotting</a>
<ul>
<li class="chapter" data-level="5.1.2.1" data-path="id_05_model_interpretation.html"><a href="id_05_model_interpretation.html#pv-pf-combined-data-set-4"><i class="fa fa-check"></i><b>5.1.2.1</b> Pv + Pf combined data set</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="id_05_model_interpretation.html"><a href="id_05_model_interpretation.html#clustering-of-predicted-positives"><i class="fa fa-check"></i><b>5.2</b> Clustering of predicted positives</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="id_05_model_interpretation.html"><a href="id_05_model_interpretation.html#analysis-5"><i class="fa fa-check"></i><b>5.2.1</b> Analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="id_05_model_interpretation.html"><a href="id_05_model_interpretation.html#comparison-of-amino-acid-frequencies-between-species"><i class="fa fa-check"></i><b>5.3</b> Comparison of amino acid frequencies between species</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="id_05_model_interpretation.html"><a href="id_05_model_interpretation.html#analysis-6"><i class="fa fa-check"></i><b>5.3.1</b> Analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="id_05_model_interpretation.html"><a href="id_05_model_interpretation.html#variable-importance"><i class="fa fa-check"></i><b>5.4</b> Variable importance</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="id_05_model_interpretation.html"><a href="id_05_model_interpretation.html#analysis-7"><i class="fa fa-check"></i><b>5.4.1</b> Analysis</a>
<ul>
<li class="chapter" data-level="5.4.1.1" data-path="id_05_model_interpretation.html"><a href="id_05_model_interpretation.html#pv-data-set-4"><i class="fa fa-check"></i><b>5.4.1.1</b> Pv data set</a></li>
<li class="chapter" data-level="5.4.1.2" data-path="id_05_model_interpretation.html"><a href="id_05_model_interpretation.html#pv-pf-combined-data-set-5"><i class="fa fa-check"></i><b>5.4.1.2</b> Pv + Pf combined data set</a></li>
</ul></li>
<li class="chapter" data-level="5.4.2" data-path="id_05_model_interpretation.html"><a href="id_05_model_interpretation.html#plotting-4"><i class="fa fa-check"></i><b>5.4.2</b> Plotting</a>
<ul>
<li class="chapter" data-level="5.4.2.1" data-path="id_05_model_interpretation.html"><a href="id_05_model_interpretation.html#pv-data-set-5"><i class="fa fa-check"></i><b>5.4.2.1</b> Pv data set</a></li>
<li class="chapter" data-level="5.4.2.2" data-path="id_05_model_interpretation.html"><a href="id_05_model_interpretation.html#pv-pf-combined-data-set-6"><i class="fa fa-check"></i><b>5.4.2.2</b> Pv + Pf combined data set</a></li>
</ul></li>
<li class="chapter" data-level="5.4.3" data-path="id_05_model_interpretation.html"><a href="id_05_model_interpretation.html#comparison-of-top-10-important-variables"><i class="fa fa-check"></i><b>5.4.3</b> Comparison of top 10 important variables</a>
<ul>
<li class="chapter" data-level="5.4.3.1" data-path="id_05_model_interpretation.html"><a href="id_05_model_interpretation.html#venn-diagram"><i class="fa fa-check"></i><b>5.4.3.1</b> Venn diagram</a></li>
<li class="chapter" data-level="5.4.3.2" data-path="id_05_model_interpretation.html"><a href="id_05_model_interpretation.html#top-10-variables-from-combined-model"><i class="fa fa-check"></i><b>5.4.3.2</b> Top 10 variables from combined model</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="id_06_top_candidate_comparisons.html"><a href="id_06_top_candidate_comparisons.html"><i class="fa fa-check"></i><b>6</b> Top candidate comparisons</a>
<ul>
<li class="chapter" data-level="6.1" data-path="id_06_top_candidate_comparisons.html"><a href="id_06_top_candidate_comparisons.html#clustering-analysis"><i class="fa fa-check"></i><b>6.1</b> Clustering analysis</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="id_06_top_candidate_comparisons.html"><a href="id_06_top_candidate_comparisons.html#analysis-8"><i class="fa fa-check"></i><b>6.1.1</b> Analysis</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="id_06_top_candidate_comparisons.html"><a href="id_06_top_candidate_comparisons.html#gene-ontology-analysis"><i class="fa fa-check"></i><b>6.2</b> Gene ontology analysis</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="id_06_top_candidate_comparisons.html"><a href="id_06_top_candidate_comparisons.html#analysis-9"><i class="fa fa-check"></i><b>6.2.1</b> Analysis</a></li>
<li class="chapter" data-level="6.2.2" data-path="id_06_top_candidate_comparisons.html"><a href="id_06_top_candidate_comparisons.html#plotting-5"><i class="fa fa-check"></i><b>6.2.2</b> Plotting</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="id_06_top_candidate_comparisons.html"><a href="id_06_top_candidate_comparisons.html#candidate-antigen-characterization"><i class="fa fa-check"></i><b>6.3</b> Candidate antigen characterization</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="id_06_top_candidate_comparisons.html"><a href="id_06_top_candidate_comparisons.html#processing-table"><i class="fa fa-check"></i><b>6.3.1</b> Processing table</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="id_07_summary_of_candidates.html"><a href="id_07_summary_of_candidates.html"><i class="fa fa-check"></i><b>7</b> Summary of candidates</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Malaria vaccine antigen identification for <em>Plasmodium vivax</em></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="page-not-found" class="section level1">
<h1>Page not found</h1>
<p>The page you requested cannot be found (perhaps it was moved or renamed).</p>
<p>You may want to try searching to find the page's new location, or use
the table of contents to find the page you are looking for.</p>
</div>
            </section>

          </div>
        </div>
      </div>


    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": false,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": [["main_notebook.pdf", "PDF"]],
"search": false,
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
